fastembed/common/onnx_model.py

```diff
@@ -52,6 +52,13 @@ def _load_onnx_model(
         model_path = model_dir / model_file
         # List of Execution Providers: https://onnxruntime.ai/docs/execution-providers
 
+        if cuda and providers is not None:
+            warnings.warn(
+                "You have specified both `cuda=True` and `providers`. This creates a conflict because `providers` takes precedence. "
+                "Avoid specifying `providers` and use `cuda=True` at the same time.",
+                category=UserWarning,
+            )
```

Comment:
It seems to be too verbose, let's make it a bit shorter and give a piece of advice instead?

```suggestion
        if cuda and providers is not None:
            warnings.warn(
                "`cuda` and `providers` are mutually exclusive parameters. If you want to utilise multiple GPUs, use `cuda` and `device_ids`."
                category=UserWarning,
                stacklevel=6
            )
```